### ISSUE TITLE
canal network running out of ip addresses

### ISSUE THREAD

**Issue by faraazkhan** [projectcalico/canal#36]
Hi, 

So I decided to test canal out on our clusters this weekend and am running into a simple issue (that maybe a configuration error), but I could not figure it out. Hoping to get some help here. 

Symptoms:

Multiple restarts of the canal pods running on nodes (the pods running on master seem fine):
```
kubectl get po -n kube-system | grep canal                                                                                                              master ✭ ✱ ◼
canal-node-0dy92                                                       3/3       Running             22         23h
canal-node-ajf16                                                       3/3       Running             0          23h
canal-node-bhvvv                                                       3/3       Running             44         23h
canal-node-ewjhn                                                       3/3       Running             0          23h
canal-node-l60ze                                                       3/3       Running             69         23h
canal-node-ueo38                                                       3/3       Running             0          23h
```

Heapster and other multiple pods failing with error message like: 
```
  2h		14s		755	{kubelet ip-172-20-68-62.us-west-2.compute.internal}			Warning		FailedSync	Error syncing pod, skipping: failed to "SetupNetwork" for "heapster-x32bq_kube-system" with SetupNetworkError: "Failed to setup network for pod \"heapster-x32bq_kube-system(9815cd0c-f0dd-11e6-87d7-06cccf320c3d)\" using network plugins \"cni\": no IP addresses available in network: canal; Skipping pod"
```
Two out of three nodes are now marked by K8S as not ready and unreachable. 
```
NAME                                           STATUS         AGE
ip-172-20-112-137.us-west-2.compute.internal   Ready,master   23h
ip-172-20-125-205.us-west-2.compute.internal   NotReady       23h
ip-172-20-43-216.us-west-2.compute.internal    NotReady       23h
ip-172-20-53-41.us-west-2.compute.internal     Ready,master   23h
ip-172-20-68-62.us-west-2.compute.internal     Ready          23h
ip-172-20-94-83.us-west-2.compute.internal     Ready,master   23h
```

Found the following logs looking through one of the canal pods with restarts:

```
2017-02-12 04:24:03,841 [INFO][40/13] calico.felix.fiptables 603: Transaction included a refresh, re-applying our inserts and deletions.
2017-02-12 04:24:14,690 [INFO][40/14] calico.felix.fiptables 547: Refreshing all our chains
2017-02-12 04:24:14,695 [INFO][40/14] calico.felix.fiptables 566: IptablesUpdater<v6-filter,queue_len=0,live=True,msg=None,time=9.492s> Successfully processed iptables updates.
2017-02-12 04:24:14,696 [INFO][40/14] calico.felix.fiptables 603: Transaction included a refresh, re-applying our inserts and deletions.
2017-02-12 04:24:19.169 [INFO][31] config_batcher.go 107: Sending config update global: map[ReportingIntervalSecs:0 ClusterGUID:bb235932a9e84c968d5d8d09b0cf7108 InterfacePrefix:cali LogSeverityFile:none LogSeverityScreen:info LogFilePath:none IpInIpEnabled:false], host: map[marker:created DefaultEndpointToHostAction:RETURN].
2017-02-12 04:24:19.169 [INFO][31] event_buffer.go 185: Possible config update. global=map[InterfacePrefix:cali LogSeverityFile:none LogSeverityScreen:info LogFilePath:none IpInIpEnabled:false ReportingIntervalSecs:0 ClusterGUID:bb235932a9e84c968d5d8d09b0cf7108] host=map[marker:created DefaultEndpointToHostAction:RETURN]
2017-02-12 04:24:19.169 [INFO][31] config_params.go 167: Merging in config from datastore (global): map[ClusterGUID:bb235932a9e84c968d5d8d09b0cf7108 InterfacePrefix:cali LogSeverityFile:none LogSeverityScreen:info LogFilePath:none IpInIpEnabled:false ReportingIntervalSecs:0]
2017-02-12 04:24:19.169 [INFO][31] config_params.go 209: Parsing value for FelixHostname: ip-172-20-94-83 (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for FelixHostname: ip-172-20-94-83 (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 209: Parsing value for EtcdCertFile:  (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for EtcdCertFile:  (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 209: Parsing value for EtcdScheme:  (from environment variable)
2017-02-12 04:24:19.170 [ERROR][31] config_params.go 226: Failed to parse config parameter EtcdScheme; value "": unknown option (source environment variable)
2017-02-12 04:24:19.170 [ERROR][31] config_params.go 233: Replacing invalid value with default value for EtcdScheme: http
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for EtcdScheme: http (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 209: Parsing value for EtcdAddr:  (from environment variable)
2017-02-12 04:24:19.170 [ERROR][31] config_params.go 226: Failed to parse config parameter EtcdAddr; value "": invalid URL authority (source environment variable)
2017-02-12 04:24:19.170 [ERROR][31] config_params.go 233: Replacing invalid value with default value for EtcdAddr: 127.0.0.1:2379
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for EtcdAddr: 127.0.0.1:2379 (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 209: Parsing value for EtcdKeyFile:  (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for EtcdKeyFile:  (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 209: Parsing value for EtcdEndpoints: http://etcd-a.internal.infra-test-debian-canal.our.domain:4001,http://etcd-b.internal.infra-test-debian-canal.our.domain:4001,http://etcd-c.internal.infra-test-debian-canal.our.domain:4001 (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for EtcdEndpoints: [http://etcd-a.internal.infra-test-debian-canal.dev.datapipe.io:4001/ http://etcd-b.internal.infra-test-debian-canal.dev.datapipe.io:4001/ http://etcd-c.internal.infra-test-debian-canal.dev.datapipe.io:4001/] (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 209: Parsing value for EtcdCaFile:  (from environment variable)
2017-02-12 04:24:19.170 [INFO][31] config_params.go 241: Parsed value for EtcdCaFile:  (from environment variable)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 209: Parsing value for LogFilePath: None (from config file)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 222: Value set to 'none', replacing with zero-value: "".
2017-02-12 04:24:19.171 [INFO][31] config_params.go 241: Parsed value for LogFilePath:  (from config file)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 209: Parsing value for LogSeverityFile: None (from config file)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 222: Value set to 'none', replacing with zero-value: "".
2017-02-12 04:24:19.171 [INFO][31] config_params.go 241: Parsed value for LogSeverityFile:  (from config file)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 209: Parsing value for MetadataAddr: None (from config file)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 222: Value set to 'none', replacing with zero-value: "".
2017-02-12 04:24:19.171 [INFO][31] config_params.go 241: Parsed value for MetadataAddr:  (from config file)
2017-02-12 04:24:19.171 [INFO][31] config_params.go 197: Ignoring unknown config param. raw name=marker
2017-02-12 04:24:19.171 [INFO][31] config_params.go 209: Parsing value for DefaultEndpointToHostAction: RETURN (from datastore (per-host))
2017-02-12 04:24:19.171 [INFO][31] config_params.go 241: Parsed value for DefaultEndpointToHostAction: RETURN (from datastore (per-host))
2017-02-12 04:24:19.171 [INFO][31] config_params.go 209: Parsing value for LogSeverityFile: none (from datastore (global))
2017-02-12 04:24:19.171 [INFO][31] config_params.go 222: Value set to 'none', replacing with zero-value: "".
2017-02-12 04:24:19.171 [INFO][31] config_params.go 241: Parsed value for LogSeverityFile:  (from datastore (global))
2017-02-12 04:24:19.171 [INFO][31] config_params.go 245: Skipping config value for LogSeverityFile from datastore (global); already have a value from config file
2017-02-12 04:24:19.171 [INFO][31] config_params.go 209: Parsing value for LogSeverityScreen: info (from datastore (global))
2017-02-12 04:24:19.171 [INFO][31] config_params.go 241: Parsed value for LogSeverityScreen: INFO (from datastore (global))
2017-02-12 04:24:19.172 [INFO][31] config_params.go 209: Parsing value for LogFilePath: none (from datastore (global))
2017-02-12 04:24:19.172 [INFO][31] config_params.go 222: Value set to 'none', replacing with zero-value: "".
2017-02-12 04:24:19.172 [INFO][31] config_params.go 241: Parsed value for LogFilePath:  (from datastore (global))
2017-02-12 04:24:19.172 [INFO][31] config_params.go 245: Skipping config value for LogFilePath from datastore (global); already have a value from config file
2017-02-12 04:24:19.172 [INFO][31] config_params.go 209: Parsing value for IpInIpEnabled: false (from datastore (global))
2017-02-12 04:24:19.172 [INFO][31] config_params.go 241: Parsed value for IpInIpEnabled: false (from datastore (global))
2017-02-12 04:24:19.172 [INFO][31] config_params.go 209: Parsing value for ReportingIntervalSecs: 0 (from datastore (global))
2017-02-12 04:24:19.172 [INFO][31] config_params.go 241: Parsed value for ReportingIntervalSecs: 0 (from datastore (global))
```

Environment:
```
Kubernetes Version: 1.4.8
Cluster Deployment Mechanism: https://github.com/kubernetes/kops with CNI networking enabled
Operating System: Debian GNU/Linux (Jessie) with https://github.com/kopeio/kubernetes-kernel
Canal Configuration Used: https://raw.githubusercontent.com/projectcalico/canal/master/k8s-install/canal.yaml (with ETCD)
```

Configuration:

```
kind: ConfigMap
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: '{"kind":"ConfigMap","apiVersion":"v1","metadata":{"name":"canal-config","namespace":"kube-system","creationTimestamp":null},"data":{"canal_iface":"","cni_network_config":"{\n    \"name\":
      \"canal\",\n    \"type\": \"flannel\",\n    \"delegate\": {\n      \"type\":
      \"calico\",\n      \"etcd_endpoints\": \"__ETCD_ENDPOINTS__\",\n      \"log_level\":
      \"info\",\n      \"policy\": {\n          \"type\": \"k8s\",\n          \"k8s_api_root\":
      \"https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__\",\n          \"k8s_auth_token\":
      \"__SERVICEACCOUNT_TOKEN__\"\n      },\n      \"kubernetes\": {\n          \"kubeconfig\":
      \"/etc/cni/net.d/__KUBECONFIG_FILENAME__\"\n      }\n    }\n}","etcd_endpoints":"http://etcd-a.internal.infra-test-debian-canal.our.domain:4001,http://etcd-b.internal.infra-test-debian-canal.our.domain:4001,http://etcd-c.internal.infra-test-debian-canal.dev.our.domain:4001","masquerade":"true"}}'
  creationTimestamp: 2017-02-11T07:35:10Z
  name: canal-config
  namespace: kube-system
  resourceVersion: "281"
  selfLink: /api/v1/namespaces/kube-system/configmaps/canal-config
  uid: 9e7b11d6-f02c-11e6-9fff-0a86d4ab4d3f
```

**Comment by caseydavenport** [comment]
@faraazkhan thanks for raising.

Can you check to see which version of `calico/cni` you're using?  A similar bug was fixed in v1.5.6 as seen here: https://github.com/projectcalico/cni-plugin/releases/tag/v1.5.6

**Comment by tmjd** [comment]
@faraazkhan It has been over a month so I am going to close this issue for now, if you are already at v1.5.6 then please reopen this issue and we can figure out the next steps to investigate.
